### PR TITLE
feat(Visualizer): Upgrade to version 2.0 viz framework

### PR DESCRIPTION
**Pull request by GordonSmith**
Convert JS to TypeScript
Bundle + Minimize JS resource
Add tutorial

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@richardkchapman please review ECL
@jbrundage please review TypeScript

File: src/WUWidget.ts [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+                if (row[2]) {
+                    edge.text(row[2]);
+                }
+                eProps.forEach(row => {
+                    if (typeof edge[row.key] === "function") {
+                        edge[row.key](row.value);
+                    }
+                });
+                return edge;
+            }
```
Comment (jbrundage) [inline src/WUWidget.ts]: Logging a meaningful error in an else condition here could potentially prevent headaches.

**Review by jbrundage — APPROVED** [review]
A few minor comments, otherwise looks good to me

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    _espWorkunit: any;
+    _wuWidgets: WUWidget[] = [];
+    _wuWidgetMap = {};
+    _wuDashSel = {};
+
+    grid: VisualizerGrid;
+
+    constructor(espUrl: string) {
+        this._espUrl = espUrl;
+        this._espWorkunit = createConnection(this._espUrl);
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply src/index.ts]: espUrl is only ever set internally - it doesn't really make sense to expose it to the user.

File: src/WUWidget.ts [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+                            .dataSource(result[0].datasource || this._metaResult.wuid())
+                            .resultName(result[0].resultname)
+                            .mappings(result[0].mappings)
+                            .filteredBy(result[0].filteredby)
+                            .properties(result[0].properties))
+                        .filteredBy(result[0].filteredby)
+                        .properties(result[0].properties)
+                        ;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply src/WUWidget.ts]: Change made as requested

File: src/WUWidget.ts (hunk 1 above)
Comment (GordonSmith) [reply src/WUWidget.ts]: Change made as requested

**Comment by GordonSmith** [comment]
@richardkchapman 2 requested changes made - should be ok to pull now.

**Pull request merged**